src/Main.elm

```diff
@@ -74,93 +74,38 @@ init _ =
 -- this is saying GotText can be a result, error, OR string?
 
 
-type AppMsg
-    = GotText (Result Http.Error String)
-
-
-type CounterMsg
+type Msg
     = Increment
     | Decrement
     | Reset
-
-
-type alias Msg =
-    { app : AppMsg
-    , counter : CounterMsg
-    }
+    | GotText (Result Http.Error String)
 
 
 {-| I think this says that update is a function that takes first a Msg type (e.g.,
 Increment) which is a second function (defined within) that also takes
 another number (Model type) and then returns a new number (Model) |
 -}
 update : Msg -> Model -> ( Model, Cmd Msg )
-update message model =
+update msg model =
     -- Handle the API request first
-    case message.app of
+    case msg of
         GotText result ->
             case result of
                 Ok fullText ->
-                    case message.counter of
-                        -- Return value +1
-                        Increment ->
-                            ( { counter = model.counter + 1
-                              , status = Success fullText
-                              }
-                            , Cmd.none
-                            )
-
-                        -- Return value -1
-                        Decrement ->
-                            ( { counter = model.counter - 1
-                              , status = Success fullText
-                              }
-                            , Cmd.none
-                            )
-
-                        -- Return 0
-                        Reset ->
-                            ( { counter = 0
-                              , status = Success fullText
-                              }
-                            , Cmd.none
-                            )
+                    ( { model | status = Success fullText }, Cmd.none )
```

Comment:
This `{model | status = ... }` syntax is a nice way to return a copy of the model where you only modify the `status` field, and leave the rest of the model the way it is.

Reply:
I remember this from the getting started / tutorial pages! I can see how it is useful in practice now.